pkg/build/api/types.go

```diff
@@ -199,6 +199,9 @@ type STIBuildStrategy struct {
 	// Image is the image used to execute the build.
 	Image string `json:"image,omitempty"`
 
+	// From is reference to an image repository from where the docker image should be pulled
+	From kapi.ObjectReference `json:"from,omitempty"`
```

Comment:
So it should be a pointer.  Since its new, image has to take priority if both are set (but we should return a validatin error if you set both).  This lets an old client work with a new resource by updating it (from would be nil when the user PUTs)